src/v/reflection/async_adl_map.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+#include "reflection/async_adl.h"
+
+#include <absl/container/btree_map.h>
+#include <absl/container/flat_hash_map.h>
+#include <absl/container/node_hash_map.h>
+
+#include <map>
+#include <unordered_map>
```

Comment:
I'd be tempted to go further. `absl` and `std` have deemed it worthy to split the implementation files.

Maybe, e.g.:
`reflection/std/map.h`
`reflection/absl/node_hash_map.h`